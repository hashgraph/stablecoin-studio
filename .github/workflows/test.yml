name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test