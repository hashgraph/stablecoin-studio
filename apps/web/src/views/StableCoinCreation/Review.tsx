import { Heading, Stack, VStack } from '@chakra-ui/react';
import type { UseFormReturn } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { SupportedWallets } from '@hashgraph/stablecoin-npm-sdk';
import DetailsReview from '../../components/DetailsReview';
import { OTHER_KEY_VALUE } from './components/KeySelector';
import { OTHER_ACCOUNT_VALUE } from './components/RoleSelector';
import { SELECTED_WALLET } from '../../store/slices/walletSlice';

interface ReviewProps {
	form: UseFormReturn;
}

const Review = (props: ReviewProps) => {
	const { form } = props;
	const { t } = useTranslation(['global', 'stableCoinCreation']);

	const wallet = useSelector(SELECTED_WALLET);

	const { getValues } = form;
	const {
		hederaTokenManagerId,
		name,
		symbol,
		initialSupply,
		supplyType,
		maxSupply,
		decimals,
		metadata,
		managementPermissions,
		isKycRequired,
		wipeKey,
		freezeKey,
		kycRequired,
		kycKey,
		pauseKey,
		cashInRoleAccount,
		burnRoleAccount,
		wipeRoleAccount,
		rescueRoleAccount,
		pauseRoleAccount,
		freezeRoleAccount,
		deleteRoleAccount,
		kycRoleAccount,
		cashInAllowance,
		cashInAllowanceType,
		manageCustomFees,
		feeScheduleKey,
		reserveAddress,
		reserveInitialAmount,
		grantKYCToOriginalSender,
		currentAccountAsProxyAdminOwner,
		proxyAdminOwnerAccount,
	} = getValues();

	const getKey = (keySelected: { value: number; label: string }, nameOtherKey: string) => {
		const { value, label } = keySelected;

		if (value === OTHER_KEY_VALUE) {
			return label + ': ' + form.watch(nameOtherKey);
		}

		return label;
	};

	const getExtraInfo = (label: string, value: string) => {
		if (label === t('stableCoinCreation:managementPermissions.cashin')) {
			if (cashInAllowanceType) {
				return `${value} - UNLIMITED`;
			} else {
				return `${value} - ${cashInAllowance}`;
			}
		} else {
			return value;
		}
	};

	const getRole = (accountSelected: { value: number; label: string }, nameOtherAccount: string) => {
		let { value, label } = accountSelected;
		if (value === OTHER_ACCOUNT_VALUE) {
			label = `${label}: ${form.watch(nameOtherAccount)}`;
		}
		return label;
	};

	const setRoleAccountInfo = (label: string, roleValue: { value: number; label: string }) => {
		const { value } = roleValue;
		if (roleValue.label !== t('stableCoinCreation:managementPermissions.none')) {
			roleDetails.push({
				label,
				value:
					value === 1
						? getExtraInfo(label, t('stableCoinCreation:managementPermissions.currentUserAccount'))
						: getExtraInfo(label, getRole(roleValue, `${label.toLowerCase()}RoleAccountOther`)),
			});
		}
	};

	const setRoleAccountInfoByKey = (
		label: string,
		roleValue: { value: number; label: string },
		keyValue: { value: number; label: string },
	) => {
		if (managementPermissions || (keyValue && keyValue.value === 2)) {
			setRoleAccountInfo(label, roleValue);
		}
	};

	const setKycRoleAccountInfoByKey = (
		label: string,
		roleValue: { value: number; label: string },
		keyValue: { value: number; label: string },
	) => {
		if (isKycRequired && keyValue && keyValue.value === 2) {
			setRoleAccountInfo(label, roleValue);
		}
	};

	const roleDetails: any[] = [];
	setRoleAccountInfo(t('stableCoinCreation:managementPermissions.cashin'), cashInRoleAccount);
	setRoleAccountInfo(t('stableCoinCreation:managementPermissions.burn'), burnRoleAccount);
	setRoleAccountInfoByKey(
		t('stableCoinCreation:managementPermissions.wipe'),
		wipeRoleAccount,
		wipeKey,
	);
	setRoleAccountInfo(t('stableCoinCreation:managementPermissions.rescue'), rescueRoleAccount);
	setRoleAccountInfoByKey(
		t('stableCoinCreation:managementPermissions.pause'),
		pauseRoleAccount,
		pauseKey,
	);
	setRoleAccountInfoByKey(
		t('stableCoinCreation:managementPermissions.freeze'),
		freezeRoleAccount,
		freezeKey,
	);
	setRoleAccountInfo(t('stableCoinCreation:managementPermissions.delete'), deleteRoleAccount);
	setKycRoleAccountInfoByKey(
		t('stableCoinCreation:managementPermissions.kyc'),
		kycRoleAccount,
		kycKey,
	);

	return (
		<VStack h='full' justify={'space-between'} pt='80px'>
			<Stack minW={400}>
				<Heading
					data-testid='title'
					fontSize='16px'
					fontWeight='600'
					mb={10}
					lineHeight='15.2px'
					textAlign={'left'}
				>
					{t('stableCoinCreation:review.title')}
				</Heading>
				<Stack as='form' spacing={9} pb='50px'>
					<DetailsReview
						title={t('stableCoinCreation:review.basicDetails')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:basicDetails.hederaTokenManager'),
								value: hederaTokenManagerId.value || '',
							},
							{
								label: t('stableCoinCreation:basicDetails.name'),
								value: name || '',
							},
							{
								label: t('stableCoinCreation:basicDetails.symbol'),
								value: symbol || '',
							},
						]}
					/>
					<DetailsReview
						title={t('stableCoinCreation:review.optionalDetails')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:optionalDetails.initialSupply'),
								value: initialSupply.toString() || '',
							},
							{
								label: t('stableCoinCreation:optionalDetails.typeSupply'),
								value: supplyType?.label || '',
							},
							{
								label: t('stableCoinCreation:optionalDetails.maxSupply'),
								value: supplyType?.value === 1 ? maxSupply.toString() : supplyType?.label,
							},
							{
								label: t('stableCoinCreation:optionalDetails.decimals'),
								value: decimals || '',
							},
							{
								label: t('stableCoinCreation:optionalDetails.metadata'),
								value: metadata || '',
							},
						]}
					/>
					<DetailsReview
						title={t('stableCoinCreation:review.managementPermissions')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:managementPermissions.admin'),
								value: t('stableCoinCreation:managementPermissions.theSmartContract'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.supply'),
								value: t('stableCoinCreation:managementPermissions.theSmartContract'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.wipe'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(wipeKey, 'wipeKeyOther'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.freeze'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(freezeKey, 'freezeKeyOther'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.kyc'),
								value: kycRequired
									? getKey(kycKey, 'kycKeyOther')
									: t('stableCoinCreation:managementPermissions.none'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.pause'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(pauseKey, 'pauseKeyOther'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.feeSchedule'),
								value: manageCustomFees
									? getKey(feeScheduleKey, 'feeScheduleKeyOther')
									: t('stableCoinCreation:managementPermissions.none'),
							},
						]}
					/>

					{roleDetails && roleDetails.length > 0 && (
						<DetailsReview
							title={t('stableCoinCreation:review.rolesAssignment')}
							titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
							details={roleDetails}
						/>
					)}

					<DetailsReview
						title={t('stableCoinCreation:managementPermissions.treasuryAccountAddress')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:managementPermissions.treasuryAccountAddress'),
								value: t('stableCoinCreation:managementPermissions.theSmartContract'),
							},
						]}
					/>

					{!reserveAddress && !reserveInitialAmount ? (
						<DetailsReview
							title={t('stableCoinCreation:proofOfReserve.title')}
							titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
							details={[
								{
									label: t('stableCoinCreation:proofOfReserve.hasPor'),
									value: t('stableCoinCreation:proofOfReserve.hasPorValueFalse'),
								},
							]}
						/>
					) : (
						<DetailsReview
							title={t('stableCoinCreation:proofOfReserve.title')}
							titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
							details={[
								{
									label: t('stableCoinCreation:proofOfReserve.hasPor'),
									value: t('stableCoinCreation:proofOfReserve.hasPorValue'),
								},
								{
									label: t('stableCoinCreation:proofOfReserve.addressPor'),
									value: reserveAddress || t('stableCoinCreation:proofOfReserve.createDataFeed'),
								},
								{
									label: t('stableCoinCreation:proofOfReserve.initialSupplyPor'),
									value: reserveInitialAmount || '-',
								},
							]}
						/>
					)}

					<DetailsReview
						title={t('stableCoinCreation:proxyAdmin.title')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:proxyAdmin.ownerAccount'),
								value: currentAccountAsProxyAdminOwner
									? t('stableCoinCreation:managementPermissions.currentUserAccount')
									: proxyAdminOwnerAccount,
							},
						]}
					/>
				</Stack>
			</Stack>
		</VStack>
	);
};

export default Review;
