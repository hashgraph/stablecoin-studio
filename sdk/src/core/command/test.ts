/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
