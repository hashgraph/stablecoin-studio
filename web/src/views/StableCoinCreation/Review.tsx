import { Heading, Stack, VStack } from '@chakra-ui/react';
import type { UseFormReturn } from 'react-hook-form';
import { useTranslation } from 'react-i18next';
import DetailsReview from '../../components/DetailsReview';
import { OTHER_KEY_VALUE } from './components/KeySelector';

interface ReviewProps {
	form: UseFormReturn;
}

const Review = (props: ReviewProps) => {
	const { form } = props;
	const { t, i18n } = useTranslation(['global', 'stableCoinCreation']);

	const { getValues } = form;
	const {
		name,
		symbol,
		autorenewAccount,
		initialSupply,
		supplyType,
		totalSupply,
		decimals,
		expirationDate,
		managementPermissions,
		adminKey,
		supplyKey,
		rescueKey,
		wipeKey,
		freezeKey,
		feeScheduleKey,
		treasuryAccountAddress,
	} = getValues();

	const formatDate = (date = '') => {
		return new Date(date).toLocaleDateString(i18n.language, {
			year: 'numeric',
			month: '2-digit',
			day: '2-digit',
		});
	};

	const getKey = (keySelected: { value: number; label: string }, nameOtherKey: string) => {
		const { value, label } = keySelected;

		if (value === OTHER_KEY_VALUE) {
			return label + ': ' + form.watch(nameOtherKey);
		}

		return label;
	};

	return (
		<VStack h='full' justify={'space-between'} pt='80px'>
			<Stack minW={400}>
				<Heading
					data-testid='title'
					fontSize='16px'
					fontWeight='600'
					mb={10}
					lineHeight='15.2px'
					textAlign={'left'}
				>
					{t('stableCoinCreation:review.title')}
				</Heading>
				<Stack as='form' spacing={9} pb='50px'>
					<DetailsReview
						title={t('stableCoinCreation:review.basicDetails')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:basicDetails.name'),
								value: name || '',
							},
							{
								label: t('stableCoinCreation:basicDetails.symbol'),
								value: symbol || '',
							},
							{
								label: t('stableCoinCreation:basicDetails.autorenewAccount'),
								value: autorenewAccount || '',
							},
						]}
					/>
					<DetailsReview
						title={t('stableCoinCreation:review.optionalDetails')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:optionalDetails.initialSupply'),
								value: initialSupply || '',
							},
							{
								label: t('stableCoinCreation:optionalDetails.typeSupply'),
								value: supplyType?.label || '',
							},
							{
								label: t('stableCoinCreation:optionalDetails.totalSupply'),
								value: supplyType?.value === 1 ? totalSupply : supplyType?.label,
							},
							{
								label: t('stableCoinCreation:optionalDetails.decimals'),
								value: decimals || '',
							},
							{
								label: t('stableCoinCreation:optionalDetails.expirationDate'),
								value: expirationDate ? formatDate(expirationDate) : '',
							},
						]}
					/>
					<DetailsReview
						title={t('stableCoinCreation:review.managementPermissions')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:managementPermissions.stableCoinAdmin'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(adminKey, 'adminKeyOther'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.supply'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(supplyKey, 'supplyKeyOther'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.rescue'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(rescueKey, 'rescueKeyOther'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.wipe'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(wipeKey, 'wipeKeyOther'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.freeze'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(freezeKey, 'freezeKeyOther'),
							},
							{
								label: t('stableCoinCreation:managementPermissions.feeSchedule'),
								value: managementPermissions
									? t('stableCoinCreation:managementPermissions.theSmartContract')
									: getKey(feeScheduleKey, 'feeScheduleKeyOther'),
							},
						]}
					/>
					<DetailsReview
						title={t('stableCoinCreation:managementPermissions.treasuryAccountAddress')}
						titleProps={{ fontWeight: 700, color: 'brand.secondary' }}
						details={[
							{
								label: t('stableCoinCreation:managementPermissions.treasuryAccountAddress'),
								value:
									!managementPermissions && supplyKey.value === OTHER_KEY_VALUE
										? treasuryAccountAddress
										: t('stableCoinCreation:managementPermissions.theSmartContract'),
							},
						]}
					/>
				</Stack>
			</Stack>
		</VStack>
	);
};

export default Review;
