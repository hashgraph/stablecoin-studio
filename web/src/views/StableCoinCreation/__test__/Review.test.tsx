import { FieldValues, useForm } from 'react-hook-form';
import { render } from '../../../test/index';
import translations from '../../../translations/en/stableCoinCreation.json';
import Review from '../Review';

jest.mock('react-hook-form', () => ({
	...jest.requireActual('react-hook-form'),
	Controller: () => <></>,
	useForm: () => ({
		getValues: () => ({
			rescueRoleAccount: { value: 1, label: '0.0.123' },
			hederaTokenManagerId: { value: 2 },
			initialSupply: 0,
			wipeKey: { value: 1, label: '0.0.123' },
			freezeKey: { value: 1, label: '0.0.123' },
			kycKey: { value: 1, label: '0.0.123' },
			pauseKey: { value: 1, label: '0.0.123' },
			feeScheduleKey: { value: 1, label: '0.0.123' },
		}),
		handleSubmit: () => jest.fn(),
	}),
}));
const form = useForm<FieldValues>({
	mode: 'onChange',
});

describe(`<${Review.name} />`, () => {
	beforeEach(() => {});

	test('should render correctly', () => {
		const component = render(<Review form={form} />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should has title', () => {
		const component = render(<Review form={form} />);

		const header = component.getByTestId('title');
		expect(header).toHaveTextContent(translations.review.title);
	});
});
