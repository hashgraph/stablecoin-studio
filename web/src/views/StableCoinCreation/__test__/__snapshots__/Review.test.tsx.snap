// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-1f6llri"
  >
    <div
      class="chakra-stack css-135a0es"
    >
      <h2
        class="chakra-heading css-p7jgmy"
        data-testid="title"
      >
        Review stablecoin details prior to creation
      </h2>
      <form
        class="chakra-stack css-1wpa917"
      >
        <div
          class="css-13brihr"
          data-testid="button-box"
        >
          <div
            class="css-1i64jsd"
          >
            <p
              class="chakra-text css-ol2bzi"
              data-testid="details-review-title"
            >
              Stablecoin Details
            </p>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-0"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  HederaTokenManager impl.
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    2
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-1"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Stablecoin Name
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  />
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-2"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Stablecoin Symbol
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-13brihr"
          data-testid="button-box"
        >
          <div
            class="css-1i64jsd"
          >
            <p
              class="chakra-text css-ol2bzi"
              data-testid="details-review-title"
            >
              Supply & Metadata
            </p>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-0"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Initial supply
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    0
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-1"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Supply type
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  />
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-2"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Max supply
                </p>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-3"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Decimals
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  />
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-4"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Metadata
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-13brihr"
          data-testid="button-box"
        >
          <div
            class="css-1i64jsd"
          >
            <p
              class="chakra-text css-ol2bzi"
              data-testid="details-review-title"
            >
              Permissions
            </p>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-0"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Admin
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    HederaTokenManager Smart Contract
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-1"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Supply
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    HederaTokenManager Smart Contract
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-2"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Wipe
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    0.0.123
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-3"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Freeze
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    0.0.123
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-4"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  KYC
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    None
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-5"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Pause
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    0.0.123
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-6"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Fee Schedule
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    None
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-13brihr"
          data-testid="button-box"
        >
          <div
            class="css-1i64jsd"
          >
            <p
              class="chakra-text css-ol2bzi"
              data-testid="details-review-title"
            >
              Roles assignment
            </p>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-0"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Cash in
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    Current user account - undefined
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-1"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Burn
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    Current user account
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-2"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Rescue
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    Current user account
                  </p>
                </div>
              </div>
              <hr
                aria-orientation="horizontal"
                class="chakra-divider css-zaaihy"
                title="divider"
              />
            </div>
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-3"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Delete
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    Current user account
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-13brihr"
          data-testid="button-box"
        >
          <div
            class="css-1i64jsd"
          >
            <p
              class="chakra-text css-ol2bzi"
              data-testid="details-review-title"
            >
              Treasury Account Address
            </p>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-0"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Treasury Account Address
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    HederaTokenManager Smart Contract
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-13brihr"
          data-testid="button-box"
        >
          <div
            class="css-1i64jsd"
          >
            <p
              class="chakra-text css-ol2bzi"
              data-testid="details-review-title"
            >
              Proof-of-Reserve
            </p>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-0"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Has Proof-of-Reserve
                </p>
                <div
                  class="chakra-stack css-1c8ysgd"
                >
                  <p
                    class="chakra-text css-mgupmj"
                  >
                    False
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-13brihr"
          data-testid="button-box"
        >
          <div
            class="css-1i64jsd"
          >
            <p
              class="chakra-text css-ol2bzi"
              data-testid="details-review-title"
            >
              Proxy Admin
            </p>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-0"
            >
              <div
                class="css-gg4vpm"
                data-testid="details-review-detail-0"
              >
                <p
                  class="chakra-text css-9s6br4"
                >
                  Owner account
                </p>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
